client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/ContactDateCard/InteractionCard/ExcelUploader/ContactUploader.tsx

```diff
@@ -50,33 +55,44 @@ const ContactUploader = ({contactEvent, onSave}:ExcelUploaderProps) => {
                 investigation: epidemiologyNumber,
                 user: userId
             })
-            axios.post('/contactedPeople/excel', {contactEvent, contacts})
-                .then((result) => {
-                    if (result.data.includes(duplicateIdsErrorMsg)) {
-                        handleDuplicateIdsError(result.data.split(':')[1], userId, epidemiologyNumber);
-                    } else {
-                        logger.info({
-                            service: Service.CLIENT,
-                            severity: Severity.LOW,
-                            workflow: 'Saving Contacted People Excel',
-                            step: 'contacted people excel was saved successfully',
-                            investigation: epidemiologyNumber,
-                            user: userId
-                        })
-                        onSave();
-                    }
-                })
-                .catch(error => {
-                    logger.error({
-                        service: Service.CLIENT,
-                        severity: Severity.LOW,
-                        workflow: 'Saving Contacted People Excel',
-                        step: `got error from server: ${error}`,
-                        investigation: epidemiologyNumber,
-                        user: userId
-                    })
-                    alertError('שגיאה בשמירת הנתונים');
+            const excelContactsIds = contacts.map((contact) => contact.identificationNumber)
+                                             .filter(id => id);
+            if(contacts.some((contact) => contact.contactStatus == COMPLETE_CONTACT_QUESTIONING_STATUS)) {
+                logger.error({
+                    service: Service.CLIENT,
+                    severity: Severity.LOW,
+                    workflow: 'Saving Contacted People Excel',
+                    step: `trying to save excel with complete status contact`,
+                    investigation: epidemiologyNumber,
+                    user: userId
                 })
+                alertError('לא ניתן לטעון מגע בסטטוס הושלם התחקור, שנו סטטוס זה ונסו שוב');
+            }
+            else if(!checkDuplicateIds(excelContactsIds.concat(existingContactsIds))) {
```

Comment:
I dont know but maybe in order to create misunderstanding you should use [...arr1, ...arr2] because the concat function can make somone new to think that .concat changes one of the arrays

Reply:
but its not mutating the array, like almost any array functions in java script concat is immutable, i think that your option isn't better, as same i can say that someone new wont be familiar with concat immutability some new might not know what spread does.
if its a common standart i will change it, but i think its more clear and readable the way it is now